forge/test/operators/pytorch/test_all.py

```diff
@@ -0,0 +1,377 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+
+# SPDX-License-Identifier: Apache-2.0
+
+# Tests for testing all pytorch operators
+
+
+# Examples
+# pytest -svv forge/test/operators/pytorch/test_all.py::test_unique --collect-only
+# TEST_ID='no_device-ge-FROM_HOST-None-(1, 2, 3, 4)-Float16_b-HiFi4' pytest -svv forge/test/operators/pytorch/test_all.py::test_single
+# OPERATORS=add,div FILTERS=HAS_DATA_FORMAT,QUICK DEV_DATA_FORMATS=Float16_b,Int8 MATH_FIDELITIES=HiFi4,HiFi3 RANGE=5 pytest -svv forge/test/operators/pytorch/test_all.py::test_query --collect-only
+
+# pytest -svv forge/test/operators/pytorch/test_all.py::test_plan
+# pytest -svv forge/test/operators/pytorch/test_all.py::test_failed
+# pytest -svv forge/test/operators/pytorch/test_all.py::test_skipped
+# pytest -svv forge/test/operators/pytorch/test_all.py::test_fatal
+# pytest -svv forge/test/operators/pytorch/test_all.py::test_not_implemented
+# pytest -svv forge/test/operators/pytorch/test_all.py::test_data_mismatch
+# pytest -svv forge/test/operators/pytorch/test_all.py::test_unsupported_df
+# pytest -svv forge/test/operators/pytorch/test_all.py::test_custom
+
+
+import os
+import pytest
+import forge
+
+from loguru import logger
+
+from test.operators.utils import DeviceUtils
+from test.operators.utils import InputSource
+from test.operators.utils import TestVector
+from test.operators.utils import TestCollection
+from test.operators.utils import TestCollectionCommon
+from test.operators.utils import TestPlanScanner
+from test.operators.utils import FailingReasons
+
+
+class TestVerification:
+
+    DRY_RUN = False
+    # DRY_RUN = True
+
+    @classmethod
+    def verify(cls, test_vector: TestVector, test_device):
+        if cls.DRY_RUN:
+            # pytest.skip("Dry run")
+            return
+        test_vector.verify(test_device)
+
+
+class TestParamsData:
+
```

Comment:
Can we add just one or two sentences to explain each class, because our framework is growing. For example: This is a parameter manager that collects input from environment variables to determine the tests that should run, which operators are involved, and which devices and data formats to use. It can create both test collections and filtering lambdas...

Something like that just to give an overview?
Apart from that LGTM!

Reply:
Included code comments